modules/@apostrophecms/schema/index.js

```diff
@@ -772,10 +776,10 @@ module.exports = {
           } else if ((item && ((typeof item._id) === 'string'))) {
             const result = results.find(doc => (doc._id === item._id));
             if (result) {
-              if (field.relationship) {
-                result._relationship = {};
-                if (item && ((typeof item._relationship === 'object'))) {
-                  await self.convert(req, field.relationship, item._relationship || {}, result._relationship);
+              if (field.schema) {
```

Comment:
The `schema` property is an implementation detail, computed from the `fields` property. It's in array form so that it is compatible with the schema module. The same thing is done at the end with the schema resulting from `fields` for any doc type. At a future point, we might rewrite the schema module to deal natively with an object rather than an array, but that would be ambitious for 3.x given our timeline.